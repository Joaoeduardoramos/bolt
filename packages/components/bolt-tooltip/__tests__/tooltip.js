import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../tooltip.schema.yml'));
const { direction, noWrap, spacing } = schema.properties;

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

const timeout = 60000;

describe('<bolt-tooltip> Component', () => {
  afterAll(async () => {
    await stopTwigRenderer();
  }, timeout);

  // Basic Usage
  test('Basic usage', async () => {
    const results = await renderTwig('@bolt-components-tooltip/tooltip.twig', {
      trigger: {
        type: 'text',
        text: 'Trigger text',
        icon: {
          name: 'info-open',
        },
      },
      content: 'Tooltip content.',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  direction.enum.forEach(async directionChoice => {
    test(`Direction of the tooltip bubble: ${directionChoice}`, async () => {
      const results = await renderTwig(
        '@bolt-components-tooltip/tooltip.twig',
        {
          trigger: {
            type: 'button',
            text: 'Trigger text',
            icon: {
              name: 'info-open',
            },
          },
          content: 'Tooltip content.',
          direction: directionChoice,
        },
      );
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  noWrap.enum.forEach(async noWrapChoice => {
    test(`No wrapping for tooltip content: ${noWrapChoice}`, async () => {
      const results = await renderTwig(
        '@bolt-components-tooltip/tooltip.twig',
        {
          trigger: {
            type: 'text',
            text: 'Trigger text',
            icon: {
              name: 'info-open',
            },
          },
          content:
            'Tooltip content. This text should wrap if noWrap is set to false, otherwise it should not wrap.',
          noWrap: noWrapChoice,
        },
      );
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  spacing.enum.forEach(async spacingChoice => {
    test(`Direction of the tooltip bubble: ${spacingChoice}`, async () => {
      const results = await renderTwig(
        '@bolt-components-tooltip/tooltip.twig',
        {
          trigger: {
            type: 'text',
            text: 'Trigger text',
            icon: {
              name: 'info-open',
            },
          },
          content: 'Tooltip content.',
          spacing: spacingChoice,
        },
      );
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });
});
