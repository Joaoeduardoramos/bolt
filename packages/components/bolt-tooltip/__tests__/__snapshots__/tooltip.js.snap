// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-tooltip> Component Basic usage 1`] = `
<bolt-tooltip trigger-type="text"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              content="Tooltip content."
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Bolt component content 1`] = `
<bolt-tooltip trigger-type="text"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              content="&lt;bolt-ul&gt;
        &lt;bolt-li&gt;1&lt;/bolt-li&gt;
        &lt;bolt-li&gt;2&lt;/bolt-li&gt;
        &lt;bolt-li&gt;3&lt;/bolt-li&gt;
        &lt;bolt-li&gt;4&lt;/bolt-li&gt;
      &lt;bolt-ul&gt;"
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Bolt component content 2`] = `
<bolt-tooltip trigger-type="text"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              content="This is the tooltip content with html &lt;a href=&quot;example.com&quot;&gt;link&lt;/a&gt;."
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Direction of the tooltip bubble: down 1`] = `
<bolt-tooltip trigger-type="button"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              position-vert="down"
              content="Tooltip content."
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Direction of the tooltip bubble: medium 1`] = `
<bolt-tooltip trigger-type="text"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              spacing="medium"
              content="Tooltip content."
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Direction of the tooltip bubble: none 1`] = `
<bolt-tooltip trigger-type="text"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              spacing="none"
              content="Tooltip content."
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Direction of the tooltip bubble: small 1`] = `
<bolt-tooltip trigger-type="text"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              spacing="small"
              content="Tooltip content."
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Direction of the tooltip bubble: up 1`] = `
<bolt-tooltip trigger-type="button"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              position-vert="up"
              content="Tooltip content."
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Direction of the tooltip bubble: xsmall 1`] = `
<bolt-tooltip trigger-type="text"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              spacing="xsmall"
              content="Tooltip content."
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component No wrapping for tooltip content: false 1`] = `
<bolt-tooltip trigger-type="text"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              content="Tooltip content. This text should wrap if noWrap is set to false, otherwise it should not wrap."
>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component No wrapping for tooltip content: true 1`] = `
<bolt-tooltip trigger-type="text"
              trigger-text="Trigger text"
              trigger-icon-name="info-open"
              no-wrap
              content="Tooltip content. This text should wrap if noWrap is set to false, otherwise it should not wrap."
>
</bolt-tooltip>
`;
